### ISSUE TITLE
Some bugs in LuaHBTeX or LuaOTFload with mode=harf?

### ISSUE THREAD

**Issue by Thomas01234** [latex3/luaotfload#141]
Please try the following MWEs with LuaHBTeX:

```
% Plain LuaHBTeX
\input luaotfload.sty

\font\twentyfiverm="MinionPro-Regular.otf:%
   mode=harf;script=latn;language=DFLT;+tlig" at%
   \directlua{tex.print(10*1.2^5)} pt% PROBLEM with Minion and mode=harf!

%\font\twentyfiverm="MinionPro-Regular.otf:%
%mode=node;script=latn;language=DFLT;+tlig" at%
%\directlua{tex.print(10*1.2^5)} pt% mode=node OK!

%\font\twentyfiverm="lmroman10-regular.otf:%
%mode=harf;script=latn;language=DFLT;+tlig" at%
%\directlua{tex.print(10*1.2^5)} pt% mode=harf with Latin Modern OK!

\baselineskip\directlua{tex.print(10*1.2^6)}pt
\twentyfiverm

\font\twentyfivemath="latinmodern-math.otf:mode=base;script=math" at%
   \directlua{tex.print(10*1.2^5)} pt

\textfont1=\twentyfivemath

\Umathcode`+="2"1`+
\Umathcode`-="2"1`−
\Umathcode`±="2"1`±
\Umathcode`=="3"1`=
\Umathcode`(="4"1`(     \Udelcode`(="1`(
\Umathcode`)="5"1`)     \Udelcode`)="1`)
\Umathcode`𝕚="7"1`𝕚     \let\Bbbi=𝕚

Die Addition von komplexen Zahlen:
%
$$ a + b𝕚 ± (c + d𝕚) = a ± c + (b ± d)𝕚 $$% PROBLEM!!!!!!!!!!!!!!!
%
Multiplikation:% this word is gobbled!
%
$$ (a + b𝕚)(c + d𝕚) = ac - bd + (ad + bc)𝕚 $$

\bye
```
As shown in the picture, the word "Multiplikation" is gobbled. What is wrong? In bigger documents with many formulae LuaHBTeX even crashes without any error message but only when I use Minion as the textfont with `mode=harf`.

When I change the mode into mode=node or when I change the font into lmroman10-regular the missing word appears.

Here is another MWE. It compares the math typesetting with the options mode=harf (wrong) and mode=base(OK).

```
% Plain LuaHBTeX
\input luaotfload.sty

\font\twentyfiverm="lmroman10-regular.otf:%
mode=harf;script=latn;language=DFLT;+tlig" at%
\directlua{tex.print(10*1.2^5)} pt
\baselineskip\directlua{tex.print(10*1.2^6)}pt
\twentyfiverm

\font\twentyfivemath="latinmodern-math.otf:mode=harf;script=math" at%
   \directlua{tex.print(10*1.2^5)} pt

\textfont1=\twentyfivemath

\Umathcode`3="7"1`3
\Umathcode`4="7"1`4

WRONG (mode=harf):
$$ 3 \over 4 $$

\font\twentyfivemath="latinmodern-math.otf:mode=base;script=math" at%
\directlua{tex.print(10*1.2^5)} pt

\textfont1=\twentyfivemath

OK (mode=base):
$$ 3 \over 4 $$

\bye
```
<img width="397" alt="harftest" src="https://user-images.githubusercontent.com/63255189/78586557-e3e41280-783b-11ea-9b99-39d82b2a4bbc.png">
<img width="237" alt="harftest2" src="https://user-images.githubusercontent.com/63255189/78586569-e7779980-783b-11ea-93bd-df0545ef8ccf.png">

I have posted this first on: https://tex.stackexchange.com/questions/537127/some-bugs-in-luahbtex-or-luaotfload-with-mode-harf

**Comment by zauguin** [comment]
I'm looking into the first issue, but the second one won't be fixed. Math typesetting is not supported with `mode=harf`. (Multiple reasons for that: First, it's basically useless because the math layout in done by the engine anyway. Also most math paramteres either can't be queried though luaharfbuzz or are not available through HarfBuzz interfaces at all)

**Comment by u-fischer** [comment]
@zauguin I can reproduce the first in miktex (texlive doesn't find the minion font) . So if needed I could proved the lua.

~~~~
\documentclass{article}
\usepackage{fontspec}
\setmainfont{MinionPro-Regular.otf}[Renderer=Harfbuzz]
\begin{document}
Die Addition von komplexen Zahlen:

\[ a \]

Multiplikation:% this word is gobbled!

\end{document}
~~~~

gives 

![image](https://user-images.githubusercontent.com/4047173/78588116-5950e280-783e-11ea-8621-2bf5000ac767.png)

**Comment by zauguin** [comment]
Should be fixed in the dev branch.

**Comment by zauguin** [comment]
@u-fischer Could you test the new version?

**Comment by u-fischer** [comment]
@zauguin  looks good. This probably mean that there should be a release in the next future. Anything else in the pipeline?

**Comment by zauguin** [comment]
@u-fischer There are two things basically: I changed the default ignorable code so that we can change the list of default ignorabes. So we could change U+AD to show up to allow it's use in discretionaries/as prehyphenchar. The question is if we want to wait for a released LaTeX change first to translate U+AD to \- to avoid hyphens from showing up in the input.

There is also a another change I am currently testing to allow nodes to be mostly ignored by both shapers which would be very useful e.g. for letterspacing/luacolor/lua-ul etc., but it still needs some testing.

**Comment by zauguin** [comment]
Fixed in v3.13.
